ui/src/ImageForm/components/ImageForm.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import Form from 'react-bootstrap/Form';
+import FormGroup from 'react-bootstrap/FormGroup';
+import Dropdown from 'react-bootstrap/Dropdown';
+import './ImageForm.css';
+import DropBtn from '../../CustomElements/DropBtn.js'
+import APICalls from '../../CustomElements/APICalls'
+
+class ImageForm extends React.Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            imageList: [],
+            selectedValue: '',
+        }
+        this.API = new APICalls()
+    }
+
+    async updateConfigmap() {
+        var data = await this.API.APIGet(this.API._CMPATH)
+        this.setState({userCM: data, selectedValue: data['last_selected_image']}, () => {this.updateImages()})
+    }
+
+    async updateImages(){
+        var data = await this.API.APIGet(this.API._IMAGEPATH)
+        //this.setState({imageList: data}, () => console.log("Taking image"))
+        this.setState({imageList: data}, () => this.isImageCorrect())
+    }
+
+    componentDidMount() {
+        this.updateConfigmap()
+    }
+    
+    isImageCorrect() {
+        console.log("Entered image check function")
+        for(var i = 0; i < this.state.imageList.length; i++) {
+            if (this.state.imageList[i] === this.state.selectedValue) {
+                return
+            }
+        }
+        if (this.state.imageList[0]) {
+            this.setState({selectedValue: this.state.imageList[0]}, console.log("Set default image"))
+            this.postChange(this.state.imageList[0])
+        }     
+    }
+
+    async postChange(text) {
+        if (typeof text !== "string") {
+            text = text.target.text
+        }
+        this.setState({selectedValue: text})
+        var json = JSON.stringify({last_selected_image: text})
+        await this.API.APIPost(this.API._CMPATH, json)
+        console.log("Sent image")
+        this.updateConfigmap() // might delete if considered not necessary, but would prevent realtime CM update
```

Comment:
Consider removing this since re-loading user CM and images on every click on the dropdown might be an overkill